Data/Table/Row/HasRows.idr

```diff
@@ -61,3 +61,36 @@ init : (tbl : Table schema)
 init tbl k =
     let Val _ = length tbl in
     dropRows tbl $ complement k
+
+public export
+(|+|) :
+     (table1 : Table schema1)
+  -> (0 nrows1 : table1 `HasRows` n)
+  => (table2 : Table schema2)
+  -> (0 nrows2 : table2 `HasRows` n)
+  => Table (schema1 |+| schema2)
+(      [<]      |+|       [<]     )
+                        {nrows1 = EmptyTable      , nrows2 = EmptyTable      }
+                                   = [<]
+
+((tbl1 :< rec1) |+| (tbl2 :< rec2))
+                        {nrows1 = SnocTable nrows1, nrows2 = SnocTable nrows2}
+                                   = (tbl1 |+| tbl2) :< (rec1 |+| rec2)
+
+(      [<]      |+| (_    :< _)) {nrows1 = EmptyTable, nrows2 = _} impossible
+((_    :< _)    |+| [<]        ) {nrows1 = _, nrows2 = EmptyTable} impossible
+
+public export
+0
+hConcatHasRows :
```

Comment:
I don't like the name `hConcat`, but I don't know what would be better.

There's a certain `zip`-like quality to it, at least.